easy_transformer/EasyTransformer.py

```diff
@@ -126,6 +106,39 @@ def __init__(
             logging.warning(
                 f"Invalid normalization_type passed in {self.cfg.normalization_type}"
             )
+
+    def __init__(
+        self,
+        cfg,
+        tokenizer=None,
+        move_to_device=True,
+    ):
+        """
+        Model initialization. Note that if you want to load the model from pretrained weights, you should use the EasyTransformer.from_pretrained() class method instead of this one.
+
+        cfg Union[EasyTransformerConfig, Dict]: The config to use for the
+            model.
+        tokenizer (*optional): The tokenizer to use for the model. If not
+            provided, it is inferred from cfg.tokenizer_name or initialized to None.
+            If None, then the model cannot be passed strings, and d_vocab must be explicitly set.
+        move_to_device (bool): Whether to move the model to the device specified in cfg.
+            device.
+        """
+        super().__init__()
+        self.__init_config__(config=cfg)
+        self.__init_tokenizer__(tokenizer=tokenizer)
+        self.__init_d_vocab__()
+        self.__init_embeddings__()
```

Comment:
Refactoring this out feels like overkill

Reply:
as discussed on slack (at least my read, it's not 100% clear to me), seems fine if we do the style `self.cfg = ...`